client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -80,6 +81,9 @@ const Checkout = React.createClass( {
 
 		if ( this.props.cart.hasLoadedFromServer && this.props.product ) {
 			this.addProductToCart();
+			if ( this.props.couponCode ) {
```

Comment:
The same should be done here, otherwise it does not work on initial load: https://github.com/Automattic/wp-calypso/blob/57855e74886947ecb174ca2b0f43c120debc30fc/client/my-sites/upgrades/checkout/checkout.jsx#L95

Or maybe we can put it to `addProductToCart`.